models/notification.go

```diff
@@ -239,6 +256,199 @@ func (n *Notification) GetIssue() (*Issue, error) {
 	return n.Issue, err
 }
 
+// HTMLURL formats a URL-string to the notification
+func (n *Notification) HTMLURL() string {
+	if n.Comment != nil {
+		return n.Comment.HTMLURL()
+	}
+	return n.Issue.HTMLURL()
+}
+
+// NotificationList contains a list of notifications
+type NotificationList []*Notification
+
+func (nl NotificationList) getRepoIDs() []int64 {
+	var ids = make(map[int64]struct{}, len(nl))
+	for _, notification := range nl {
+		if notification.Repository != nil {
+			continue
+		}
+		if _, ok := ids[notification.RepoID]; !ok {
+			ids[notification.RepoID] = struct{}{}
+		}
+	}
+	return keysInt64(ids)
+}
+
+// LoadRepos loads repositories from database
+func (nl NotificationList) LoadRepos() (RepositoryList, error) {
+	if len(nl) == 0 {
+		return RepositoryList{}, nil
+	}
+
+	var repoIDs = nl.getRepoIDs()
+	var repos = make(map[int64]*Repository, len(repoIDs))
+	var reposList = make(RepositoryList, 0, len(repoIDs))
+	var left = len(repoIDs)
+	for left > 0 {
+		var limit = defaultMaxInSize
+		if left < limit {
+			limit = left
+		}
+		rows, err := x.
+			In("id", repoIDs[:limit]).
+			Rows(new(Repository))
+		if err != nil {
+			return nil, err
+		}
+
+		for rows.Next() {
+			var repo Repository
+			err = rows.Scan(&repo)
+			if err != nil {
+				rows.Close()
+				return nil, err
+			}
+
+			repos[repo.ID] = &repo
+			reposList = append(reposList, &repo)
+		}
+		_ = rows.Close()
+
+		left -= limit
+		repoIDs = repoIDs[limit:]
+	}
+
+	for _, notification := range nl {
+		if notification.Repository == nil {
+			notification.Repository = repos[notification.RepoID]
+		}
+	}
+	return reposList, nil
```

Comment:
If any notification had .Repository already set before calling this function, reposList may not contain that repository in it. Is that OK?

Reply:
I think this is now out of date?

Reply:
Yes.